proxy/client.go

```diff
@@ -0,0 +1,119 @@
+package proxy
+
+import (
+	"context"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/google/go-querystring/query"
+)
+
+//Client an API client to perform all operations
+type Client struct {
+	ClientAuth ClientAuth
+	httpClient *http.Client
```

Comment:
is the httpClient intended to be private? If so, can we sort it first and maybe with a new line to make it clearly separate?

Can we add docs for the public fields?

Reply:
I will add documentation for the public fields.  My intention was to keep `httpClient` private. I have just referred github client (https://github.com/moby/moby/blob/46ec8731fbce9b13e41bd85d6cea691406bba852/client/client.go#L65) and moby cli client (https://github.com/moby/moby/blob/46ec8731fbce9b13e41bd85d6cea691406bba852/client/client.go#L65). Let me know what you think.

Reply:
I think it is ok to be private, but just structuring it to be clearly obvious would be nice. When I saw it between two public fields, I thought it might be a typo

Reply:
Updated it.